hedera-node/hapi-fees/src/main/java/org/hiero/hapi/fees/FeeScheduleUtils.java

```diff
@@ -0,0 +1,106 @@
+import org.hiero.hapi.support.fees.ServiceFeeDefinition;
+import org.hiero.hapi.support.fees.ServiceFeeSchedule;
+
+import static java.util.Objects.requireNonNull;
+
+public class FeeScheduleUtils {
+    public static ExtraFeeDefinition makeExtraDef(Extra extra, long fee) {
+        return ExtraFeeDefinition.newBuilder().name(extra).fee(fee).build();
+    }
+
+    public static ExtraFeeReference makeExtraIncluded(Extra extra, int included) {
+        return ExtraFeeReference.DEFAULT
+                .copyBuilder()
+                .name(extra)
+                .includedCount(included)
+                .build();
+    }
+
+    public static ServiceFeeDefinition makeServiceFee(HederaFunctionality name, long baseFee, ExtraFeeReference... reference) {
+        return ServiceFeeDefinition.DEFAULT
+                .copyBuilder()
+                .name(name)
+                .baseFee(baseFee)
+                .extras(reference)
+                .build();
+    }
+
+    public static ServiceFeeSchedule makeService(String name, ServiceFeeDefinition... services) {
+        return ServiceFeeSchedule.DEFAULT.copyBuilder().name(name).schedule(services).build();
+    }
+
+    public static ExtraFeeDefinition lookupExtraFee(FeeSchedule feeSchedule, ExtraFeeReference ref) {
+        for (ExtraFeeDefinition def : feeSchedule.extras()) {
+            if (def.name().equals(ref.name())) {
+                return def;
+            }
+        }
+        throw new Error("Extra Fee definition not found for " + ref.name());
+    }
+
+    public static ServiceFeeDefinition lookupServiceFee(FeeSchedule feeSchedule, HederaFunctionality api) {
+        for (ServiceFeeSchedule service : feeSchedule.services()) {
+            for (ServiceFeeDefinition def : service.schedule()) {
+                if (def.name() == api) {
+                    return def;
+                }
+            }
+        }
+        throw new Error("Service definition not found for " + api.toString());
+    }
+
+    /**
+     * Validate the fee schedule. There must be
+     * no
+     * @param feeSchedule
+     */
+    public static boolean validate(FeeSchedule feeSchedule) {
+        requireNonNull(feeSchedule);
+//        System.out.println("validating " + feeSchedule);
+        for (ExtraFeeDefinition def : feeSchedule.extras()) {
+            // no negative values or greater than MAX long
+            if (def.fee() < 0) {
+                return false;
+            }
+            if (def.fee() > Long.MAX_VALUE) {
+                return false;
+            }
+        }
+
+        // all referenced extras are defined
+        for (ServiceFeeSchedule service : feeSchedule.services()) {
+            for (ServiceFeeDefinition def : service.schedule()) {
+                for (ExtraFeeReference ref : def.extras()) {
+                    try {
+                        lookupExtraFee(feeSchedule, ref);
+                    } catch (Error e) {
+                        return false;
+                    }
```

Comment:
This is a bit concerning.
We ignore RuntimeException, but catch Error?
That seems like it might not be as intended (particularly because Error throwables usually should not be caught because the JVM often cannot continue processing correctly after throwing an Error).

If we returned `null` when definitions are not found, this would just be
```suggestion
                    if (lookupExtraFee(feeSchedule, ref) == null) {
                        return false;
                    }
```